### ISSUE TITLE
log magnitude spectrogram

### ISSUE THREAD

**Issue by minsangkim142** [Kyubyong/tacotron#16]
In the paper for Tacotron, the author mentions that linear scale log magnitude spectrogram is used, however from https://github.com/Kyubyong/tacotron/blob/master/train.py#L144 we're using no log scaling in magnitude. Is it due to the zero values in spectrograms computed by Librosa library?

**Comment by Kyubyong** [comment]
No. I just neglected. Not sure if it makes any difference, though. What do you think?

**Comment by minsangkim142** [comment]
I believe the whole point of logging the target variable is to normalise it. Without log scaling the magnitude range of the spectrogram is very large hence harder for the weights to converge. Here are the two target variable (spectrogram) distribution with and without logging. What does everyone think?

![screenshot from 2017-05-30 14-05-46](https://cloud.githubusercontent.com/assets/24954419/26565669/4f87c224-4541-11e7-8161-afb45598651f.png)
![screenshot from 2017-05-30 14-07-03](https://cloud.githubusercontent.com/assets/24954419/26565670/515b5e3a-4541-11e7-86e8-c6f6e3b68efa.png)

**Comment by Kyubyong** [comment]
Does normalizing target values really help training? People care about normalizing input values (at least before the introduction of the batch normalization), but I don't fully understand if the target value normalization is helpful. But I think, of course, it's definitely worth a try.

https://stats.stackexchange.com/questions/111467/is-it-necessary-to-scale-the-target-value-in-addition-to-scaling-features-for-re

Minsang, why don't you compare them and share the result with me and others? Maybe, I'll do the experiment, too.

**Comment by Kyubyong** [comment]
On second thought, logarithm has something to do with loss scaling. The first loss is smaller than the second loss. Because we just add those two, the different scale of those two losses might be problematic. If that's the case, logarithm definitely should help.

**Comment by ljun4121** [comment]
In my opinion, normalization using linear transform will be meaningless, but taking a log can be meaningful. Most of the information in speech is distributed in the low frequency band. So taking a logarithm like mel-spectrum can spread distribution.
